test/tasks/withdrawService.test.ts

```diff
@@ -0,0 +1,342 @@
+    const daiBalanceIncludingSolver = daiBalanceMinusLeftover.add(
+      solverDaiBalance,
+    );
+    const daiFee = daiBalanceIncludingSolver.div(2);
+    const daiFeeAndQuote: GetFeeAndQuoteSellOutput = {
+      feeAmount: BigNumber.from(daiFee),
+      buyAmountAfterFee: ("unused" as unknown) as BigNumber,
+    };
+    apiMock
+      .expects("getFeeAndQuoteSell")
+      .withArgs({
+        sellToken: dai.address,
+        buyToken: toToken.address,
+        sellAmountBeforeFee: daiBalanceIncludingSolver,
+      })
+      .once()
+      .returns(Promise.resolve(daiFeeAndQuote));
+    const wethFee = wethBalance.div(1337);
+    const wethFeeAndQuote: GetFeeAndQuoteSellOutput = {
+      feeAmount: wethFee,
+      buyAmountAfterFee: BigNumber.from(1337),
+    };
+    apiMock
+      .expects("getFeeAndQuoteSell")
+      .withArgs({
+        sellToken: weth.address,
+        buyToken: toToken.address,
+        sellAmountBeforeFee: wethBalanceMinusLeftover,
+      })
+      .once()
+      .returns(Promise.resolve(wethFeeAndQuote));
+
+    const validity = 3600;
+    function assertGoodOrder(
+      order: Order,
+      sellToken: string,
+      sellAmount: BigNumber,
+      buyAmount: BigNumber,
+      feeAmount: BigNumber,
+    ) {
+      expect(order.sellToken).to.deep.equal(sellToken);
+      expect(order.buyToken).to.deep.equal(toToken.address);
+      expect(order.sellAmount).to.deep.equal(sellAmount);
+      expect(order.buyAmount).to.deep.equal(buyAmount);
+      expect(order.feeAmount).to.deep.equal(feeAmount);
+      expect(order.kind).to.deep.equal(OrderKind.SELL);
+      expect(order.receiver).to.deep.equal(receiver.address);
+      // leave a minute of margin to account for the fact that the actual
+      // time and the time at which they are compared are slightly different
+      expect(
+        Math.abs(timestamp(order.validTo) - (Date.now() / 1000 + validity)) <
+          60,
+      ).to.be.true;
+      expect(order.partiallyFillable).to.equal(false);
+    }
+    api.placeOrder = async function ({ order }: PlaceOrderQuery) {
+      switch (order.sellToken) {
+        case usdc.address: {
+          assertGoodOrder(
+            order,
+            usdc.address,
+            usdcBalanceMinusLeftover.sub(usdcFeeAndQuote.feeAmount),
+            usdcFeeAndQuote.buyAmountAfterFee,
+            usdcFeeAndQuote.feeAmount,
+          );
+          return "0xusdcOrderUid";
+        }
+        case weth.address: {
+          assertGoodOrder(
+            order,
+            weth.address,
+            wethBalanceMinusLeftover.sub(wethFeeAndQuote.feeAmount),
+            wethFeeAndQuote.buyAmountAfterFee,
+            wethFeeAndQuote.feeAmount,
+          );
+          return "0xwethOrderUid";
+        }
+        default:
+          throw new Error(
+            `Invalid sell token ${order.sellToken} in mock order`,
+          );
+      }
+    };
+
+    const updatedState = await withdrawService.withdrawAndDump({
```

Comment:
Not sure how easy it is to add a test that tests pagination? E.g. call `withdrawAndDump` at least twice to see if it resumes again from the beginning? Just an idea, feel free to ignore if this adds too much work.

Reply:
Implemented in #776.